pages/platform/kubernetes-k8s/installing-go-operator/guide.en-gb.md

```diff
@@ -0,0 +1,1390 @@
+//+kubebuilder:rbac:groups=tutorials.ovhcloud.com,resources=ovhnginxes/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=tutorials.ovhcloud.com,resources=ovhnginxes/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the OvhNginx object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *OvhNginxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// TODO(user): your logic here
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *OvhNginxReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&tutorialsv1.OvhNginx{}).
+		Complete(r)
+}
+```
+
+Next, you can update the CRD to add fields to manage the Nginx server.<br>
+Update the file `api/v1/ovhnginx_types.go`:
+
+```go
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OvhNginxSpec defines the desired state of OvhNginx
+type OvhNginxSpec struct {
+	// Number of replicas for the Nginx Pods
+	ReplicaCount int32 `json:"replicaCount"`
+	// Exposed port for the Nginx server
+	Port int32 `json:"port"`
+}
+
+// OvhNginxStatus defines the observed state of OvhNginx
+type OvhNginxStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// OvhNginx is the Schema for the ovhnginxes API
+type OvhNginx struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OvhNginxSpec   `json:"spec,omitempty"`
+	Status OvhNginxStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// OvhNginxList contains a list of OvhNginx
+type OvhNginxList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OvhNginx `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&OvhNginx{}, &OvhNginxList{})
+}
+```
+
+Then generate the manifest:
+```bash
+make manifest
+```
+
+Output should be like this:
+
+<pre class="console"><code>$ make manifests
+
+/Users/sphilipp/dev/ovh/corp/tutorials-code/nginx-go-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+</code></pre>
+
+Verify that the CRD is updated with the two fields (`port` and `replicaCount`):
```

Comment:
Verify that the CRD file **config/crd/bases/tutorials.ovhcloud.com_ovhnginxes.yaml** is updated with the two fields (`port` and `replicaCount`):